controllers/upgrade.go

```diff
@@ -109,6 +111,110 @@ func (r *RollingUpgradeContext) RotateNodes() error {
 	return nil
 }
 
+// Update last batch nodes
+func (s *RollingUpgradeContext) UpdateLastBatchNodes(batchNodes map[string]*v1alpha1.NodeInProcessing) {
```

Comment:
Can we move all these methods to `controllers/metrics.go`?

Reply:
package "api" depends on package "common". 
We can not let package "common" refer to the structures in "api.*".
Also common is used in "controllers", those methods need to access RollingUpgradeContext.

Reply:
This can stay same package @shaoxt 
just different file

Reply:
Sounds good to me.  Moved. Thanks